src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+    "LogicalJoinType": "And",
+    "Children": [
+      {
+        "TaggingGroupingCondition": {
+          "ObjectAttribute": "Name",
+          "CompareType": "Contains",
+          "CompareValue": "prd.local"
+        },
+        "TaggingGroupingConditionGroup": null
+      }
+    ]
+  }
+}' | ConvertFrom-Json
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+$taggingRule = '{"Description": null,"Enabled": true,"RuleConditionGroup": {"LogicalJoinType": "And","Children": [{"TaggingGroupingCondition": {"ObjectAttribute": "Name","CompareType": "Contains","CompareValue": ".demo"},"TaggingGroupingConditionGroup": null}]}}' | ConvertFrom-Json
+Test-SafeguardAssetTaggingRule 5 -TaggingRule $taggingRule
+
+.EXAMPLE
+$taggingRule = '{"Description": null,"Enabled": true,"RuleConditionGroup": {"LogicalJoinType": "And","Children": [{"TaggingGroupingCondition": {"ObjectAttribute": "Name","CompareType": "Contains","CompareValue": ".demo"},"TaggingGroupingConditionGroup": null}]}}' | ConvertFrom-Json
+Test-SafeguardAssetTaggingRule -AssetPartitionId 1 -Tag prod -TaggingRule $taggingRule
+#>
+function Test-SafeguardAssetTaggingRule {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Tag,
+        [Parameter(Mandatory=$true,Position=1)]
+        [PSCustomObject]$TaggingRule
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    # use the helper function Resolve-AssetPartitionId to get the asset partition id.
+    $AssetPartitionId = Resolve-AssetPartitionId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId
+    $local:TagId = (Resolve-SafeguardTagId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId $Tag)
+    if ($local:TagId) {
+        $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags/$local:TagId/TestAssetRule"
+
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core POST "$local:RelPath" -Body $TaggingRule
+    } else {
+        throw "The tag with id $TagId could not be found."
+    }
+}
+
+
+
+<#
+.SYNOPSIS
+Test an AssetAccountTaggingRule on a tag.
+
+.DESCRIPTION
+Test what an AssetAccountTaggingRule would do. 
+This can be used to verify which accounts would be assigned this the tag if the rule is set.
```

Comment:
Same...
"...would be assigned this the tag if..." Note the extra "the" after "this". Could be "...would be assigned this tag if..."